Feature/nag 34 implement global component styling

Tanaya-27 opened the pull request:
## Global Component Styling
Here is an overview of the elements that I have modulated, with their .css names and how to use them:
- `white-bg-button`: this forms the basic oblong buttons such as the shop buttons in the product card and slidey
  - call this with the button text (usage inside a <link> as it was before)
- `outline-arrow`: this is the no-background arrow which turns lightly grey (basically white) when hovered, i.e. the ones used in the banner
  - call with the appropriate icon, i.e. arrowLeft or arrowRight, and size (e.g. for banner size=20)
- `side-panel-button`: this is used for the hamburger and close button for now, but when the shopping cart is implemented it can be used there too
  - call with the actual icon used, and the on-click navigation (e.g. setIsOpen(True))
- `rating`: this is simply the format for the rating, number+star format
  - call with the rating variable and star emoji/icon
- `social-button`: this is used for the interactive socila media icons, i.e. those used in the footer and in the navbar side menu
  - call *as a link* with the reference, and include the image inside

rubenodamo reviewed (changes requested):
I've noticied that in multiple places (`Banner.jsx`, `HeroSlider.jsx`, `Navbar.jsx`, and `ProductCard.jsx`), the ShadCN `<Button>` components have been replaced with raw HTML `<button>` elements. This change moves us away from the consistent design system we've been building with ShadCN.

**Recommendation:**
- Revert these raw `<button>` elements back to the ShadCN `<Button>` component to maintain consistency across the UI.
- For the `ProductCard`, since its actions are just external links, consider following the pattern you implemented in the footer — using a `<Link>` styled with `white-bg-button` or another utility class as needed, instead of a button. I should have done this when I made the component! I never thought about it like how you did.
- Remember the buttons we are implementating are supposed to be aligned with the figma. So perhaps add some options for the yellow and green buttons, even maybe a pink too since we have added that colour to the site.
- In the ShadCN `button.jsx `class you'll find a utility called `cva` (class variance authority) that defines the variants. You can extend it like this:
```
const buttonVariants = cva(
  "inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:ring-2",
  {
    variants: {
      variant: {
        default: "bg-primary text-white hover:bg-primary/90",
        ghost: "hover:bg-accent hover:text-accent-foreground",
        // + Add your custom variant here
        whiteBg: "bg-white text-black border border-black hover:bg-gray-100",
      },
      size: {
        default: "h-10 px-4 py-2",
        sm: "h-9 px-3",
        lg: "h-11 px-8",
        icon: "h-10 w-10",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
    },
  }
);
```
_Then you can use it like this:_
```
<Button variant="whiteBg">Shop Now</Button>
```
_You can also make compound variants (e.g. `variant` + `size`) if you want more control._

Tanaya-27 commented:
- all raw `<button>` elements removed and replaced with shadCN `<Button>`s or `<Link>`s
- yellow, green, and pink buttons implemented in `button.jsx`
- two *static* sizes also implemented, i.e. not dynamic based on text or anything else (used this in the slidey for example)
- icons still as links, so CSS style remains but now called `social-icon` rather than `social-button`

#### Questions
- the shop buttons currently cannot be styled separately as they are called from the links mapping (looped through). so far i have left them as `white-bg-link` `<Link>` items, even though I have created the coloured buttons. this also means currently we cannot use different sizings for delli vs tbf blocks.
  - i can refactor this to separate them, and have the product card only take two separate shop links rather than an unlimited array. what do you think?

rubenodamo reviewed (commented):
**Summary of changes:**
- Added more variants for the buttons (based on the figma). Added: grey, green with black outline and pink with black outline buttons.
- Added hover effects to the buttons. The default ones were too dull.
  - Added `hover:scale-110`, `hover:border-2`.
  - Explicit `border-white` added to `green` and `pink` to match designs. The outlines of those were grey for some reason.
- Changed the outline variant back to how it was before as it removed the ouline from the page arrows for the product carousel, meaning the arrows were hard to see.
- Changed the button implmentation in `HeroSlider.jsx` so that it is not wrapped in `<Link>` and also added hover effects for the pagination dots.

**Testing Notes:**
- I think lines 250 - 275 can be removed from the `globals.css` file. I dont think that styling is used anymore. I could be wrong though please check and remove it if needed.
- I added on the test/page.js a way to view all the button designs. Please hover over thanm all and see if you are ok with the effects added or maybe have any other suggestions.

Otherwise it is ready to merge. I will approve the merge request but _**please read and action these items before merging**_.

Tanaya-27 commented:
- The pagination dots do not work as they should without the "outline" variant, they turn black when hovered, which doesn't look intuitive. I have creatsed a new `noOutline` variant which works in the same way the pagination dots were before
- Agreed and removed lines 250-275 in `globals.css`, ths was only kept for your original request of keeping links as non-buttons but i think what we have works better.

rubenodamo commented:
> * The pagination dots do not work as they should without the "outline" variant, they turn black when hovered, which doesn't look intuitive. I have creatsed a new `noOutline` variant which works in the same way the pagination dots were before
> 
>     * Agreed and removed lines 250-275 in `globals.css`, ths was only kept for your original request of keeping links as non-buttons but i think what we have works better.

- Shouldn't the hover effect of the pagination dots be a different colour to the actual highlighted dot (i.e. the dot that the page is currently on)? If you think this is representative of most pagination dots on the web then this can be merged.

Tanaya-27 commented:
>     * Shouldn't the hover effect of the pagination dots be a different colour to the actual highlighted dot (i.e. the dot that the page is currently on)? If you think this is representative of most pagination dots on the web then this can be merged.

I couldn't find many examples of the pagination dots but the following link shows pagination demos (mostly numbers) and the ones that change colour/style on hover seem to match the style of the current:
https://devsnap.me/css-pagination
